source/core_sntp_client.c

```diff
@@ -272,23 +271,29 @@ static SntpStatus_t validateContext( const SntpContext_t * pContext )
  * @param[in] pPacket The buffer containing the SNTP packet data
  * to send over the network.
  * @param[in] packetSize The size of data in the SNTP request packet.
+ * @param[in] timeoutMs The timeout period for retry attempts of sending
+ * SNTP request packet over the network.
  *
  * @return Returns #SntpSuccess on successful transmission of the entire
- * SNTP request packet over the network; otherwise #SntpErrorNetworkFailure
- * to indicate failure.
+ * SNTP request packet over the network; #SntpErrorNetworkFailure
+ * to indicate failure from transport interface; #SntpErrorSendTimeout if
+ * time request could not be sent over the network within the @p timeoutMs
+ * duration.
  */
 static SntpStatus_t sendSntpPacket( const UdpTransportInterface_t * pNetworkIntf,
                                     uint32_t timeServer,
                                     uint16_t serverPort,
                                     SntpGetTime_t getTimeFunc,
                                     const uint8_t * pPacket,
-                                    uint16_t packetSize )
+                                    uint16_t packetSize,
+                                    uint32_t timeoutMs )
 {
     const uint8_t * pIndex = pPacket;
     int32_t bytesSent = 0;
     SntpTimestamp_t lastSendTime;
-    uint64_t timeSinceLastSendMs;
-    bool sendError = false;
+    uint64_t elapsedTimeMs;
+    bool shouldRetry = false;
+    SntpStatus_t status;
```

Comment:
Probably a good idea to initialize the status

Reply:
Done